maven-core/src/main/java/org/apache/maven/plugin/PluginParameterExpressionEvaluator.java

```diff
@@ -85,9 +84,9 @@
 
     private Properties properties;
 
-    @Deprecated //TODO used by the Enforcer plugin
+    @Deprecated //TODO used by the Enforcer plugin (cstamas: not anymore, since 3.0.0)
     public PluginParameterExpressionEvaluator( MavenSession session, MojoExecution mojoExecution,
-                                               PathTranslator pathTranslator, Logger logger, MavenProject project,
+                                               PathTranslator pathTranslator, MavenProject project,
```

Comment:
This is a breaking change for maven-enforcer-plugin pre 3.0.0... IMO, we should just drop this (long time deprecated) constructor, as 3.x plugins are requirement for a long time, and this is Maven4.

Reply:
Will handle this in https://issues.apache.org/jira/browse/MNG-7279 to not mish-mash this PR